multichain-testing/test/ibc-transfers.ts

```diff
@@ -0,0 +1,291 @@
+  }: {
+    srcChainName: string;
+    destChainName: string;
+    acctAddr: string;
+    qty: number;
+    denom: string;
+  },
+) => {
+  const { agd, retryUntilCondition, useChain } = t.context;
+  const destChainid = useChain(destChainName).chain.chain_id;
+  const srcChainId = useChain(srcChainName).chain.chain_id;
+
+  const agoricToOsmosis =
+    starshipChainInfo[srcChainName].connections[destChainid].transferChannel;
+  t.assert(agoricToOsmosis.channelId);
+  const amount = `${qty}${denom}`;
+  // agd tx ibc-transfer transfer [src-port] [src-channel] [receiver] [amount] [flags]
+  const fundInitial = await agd.tx(
+    [
+      'ibc-transfer',
+      'transfer',
+      'transfer',
+      agoricToOsmosis.channelId,
+      acctAddr,
+      amount,
+    ],
+    {
+      chainId: srcChainId,
+      from: 'faucet',
+      yes: true,
+    },
+  );
+  t.is(fundInitial?.code, 0);
+  t.log('Success. TxHash:', fundInitial?.txhash);
+
+  const queryClient = makeQueryClient(
+    await useChain(destChainName).getRestEndpoint(),
+  );
+  const { balances } = await retryUntilCondition(
+    () => queryClient.queryBalances(acctAddr),
+    ({ balances }) => !!balances.find(x => x.denom.startsWith('ibc/')),
+    `${acctAddr} received bld from agoric`,
+  );
+
+  t.log('amtOnDest', `${qty}${balances[0].denom}`);
+  return { denom: balances[0].denom };
+};
+
+const setupSourceWallet = async (
+  t: ExecutionContext<SetupContext>,
+  { chainName }: { chainName: string },
+) => {
+  const { retryUntilCondition, useChain } = t.context;
+
+  const chaind = makeAgd({ execFileSync }).withOpts({
+    keyringBackend: 'test',
+    chainName: chainName,
+    broadcastMode: 'sync',
+  });
+  try {
+    chaind.keys.delete(keyName);
+  } catch {
+    /* key already deleted */
+  }
+  chaind.keys.add(keyName, generateMnemonic());
+  const acctAddr = chaind.keys.showAddress(keyName);
+  t.assert(acctAddr);
+  t.log('acctAddr', acctAddr);
+  // give gas funds to osmosis wallet so it can send tx
+  await useChain(chainName).creditFromFaucet(acctAddr);
+
+  const queryClient = makeQueryClient(
+    await useChain(chainName).getRestEndpoint(),
+  );
+
+  await retryUntilCondition(
+    () => queryClient.queryBalances(acctAddr),
+    ({ balances }) => !!balances.length,
+    `${acctAddr} received funds from faucet`,
+  );
+
+  return {
+    acctAddr,
+    chaind,
+  };
+};
+
+type QueryRes = { total_count: string; txs: object[] };
+const queryPackets = async (binaries: Record<string, Agd>) => {
+  const results: Record<string, { recvs: QueryRes; acks: QueryRes }> = {};
+  for (const [name, chaind] of Object.entries(binaries)) {
+    const recvs = await chaind.queryTxsByEvents(txEventQuery.msgReceivePacket);
+    const acks = await chaind.queryTxsByEvents(txEventQuery.msgAcknowledgement);
+    results[name] = {
+      recvs,
+      acks,
+    };
+  }
+  return results;
+};
+
+type ChainPacketCounts = Record<
+  string,
+  {
+    recvs: number;
+    acks: number;
+  }
+>;
```

Comment:
IIUC `interface` is preferred here. And it would help to document the key,
```suggestion
interface ChainPacketCounts {
    [chain: string]: {
        recvs: number;
        acks: number;
    }
}
```

When you do that the more concise `PacketCounts` should suffice

Reply:
Done